enterprise/server/workflow/workflow.go

```diff
@@ -319,9 +318,12 @@ func (ws *workflowService) apiKeyForWorkflow(ctx context.Context, wf *tables.Wor
 	return k, nil
 }
 
-func (ws *workflowService) parseRequest(r *http.Request) (*webhookData, error) {
-	log.Printf("Webhook body: %+v", r)
-	return &webhookData{SHA: "HEAD"}, nil
+func parseRequest(r *http.Request) (*webhook_data.WebhookData, error) {
+	if r.Header.Get("x-github-event") != "" {
+		return github.ParseRequest(r)
+	}
+	log.Printf("failed to classify Git provider from webhook request: %+v", r)
+	return nil, nil
```

Comment:
probably want to return this as unimplemented error or something so it stops the flow of the startWorkflow method.

Reply:
Done